sources/bootstrap-commands/src/main.rs

```diff
@@ -0,0 +1,367 @@
+/*!
+# Bootstrap commands
+
+bootstrap-commands ensures that Bootstrap Commands are executed as defined in the system
+settings. It is called by bootstrap-commands systemd service which runs prior to the
+execution of bootstrap-containers.
+
+Each bootstrap command is a set of commands. The service queries the API for the settings, then
+configures the system by going through all the bootstrap commands in lexicographical order and
+running all the commands inside it.
+
+## Example:
+Given a bootstrap command called `001-test-bootstrap-commands` with the following configuration:
+
+[settings.bootstrap-commands.001-test-bootstrap-commands]
+commands = [[ "apiclient", "set", "motd=helloworld"]]
+essential = true
+mode = "always"
+
+This would set `/etc/motd` to "helloworld".
+
+# Additional Information:
+Certain valid apiclient commands that work in a session may fail in bootstrap-commands due to
```

Comment:
nit: let's code-format mentions of `apiclient`, `bootstrap-commands` and `bootstrap-commands.service` , so it's clear that we're talking about specific programs or units and not something more general.
```suggestion
Certain valid `apiclient` commands that work in a session may fail in `bootstrap-commands` due to
```